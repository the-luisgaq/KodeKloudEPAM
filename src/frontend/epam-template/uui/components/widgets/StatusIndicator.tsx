import React, { forwardRef } from 'react';
import cx from 'classnames';
import type { IHasCX, Overwrite } from '@epam/uui-core';
import { settings } from '../../settings';

import css from './StatusIndicator.module.scss';

type StatusIndicatorColor = 'neutral' | 'info' | 'success' | 'warning' | 'critical';

type StatusIndicatorMods = {
    /**
     * Component color.
     * @default 'neutral'
     */
    color?: StatusIndicatorColor;
};

export interface StatusIndicatorModsOverride {}

export type StatusIndicatorCoreProps = IHasCX & {
    /**
     * Component size. If omitted, 24 size will be used.
     * @default '24'
     */
    size?: '24' | '18' | '12';
    /**
     * Fill mode, solid is used by default
     * @default 'solid'
     */
    fill?: 'solid' | 'outline';
    /** Component caption. */
    caption: string;
};

export type StatusIndicatorProps = StatusIndicatorCoreProps & Overwrite<StatusIndicatorMods, StatusIndicatorModsOverride>;

export const StatusIndicator = forwardRef<HTMLDivElement, StatusIndicatorProps>((props, ref) => {
    return (
        <div
            ref={ ref }
            className={ cx([
                css.root,
                `uui-size-${props.size || settings.statusIndicator.sizes.default}`,
                'uui-status_indicator',
                `uui-color-${props.color || 'neutral'}`,
                `uui-fill-${props.fill || 'solid'}`,
                props.cx,
            ]) }
        >
            <div className="uui-status_indicator-dot"></div>
            <p className="uui-status_indicator-caption">{props.caption}</p>
        </div>
    );
});
