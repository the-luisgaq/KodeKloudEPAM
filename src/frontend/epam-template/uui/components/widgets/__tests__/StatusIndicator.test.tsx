import React from 'react';
import { render } from '@epam/uui-test-utils';
import { StatusIndicator } from '../StatusIndicator';

describe('StatusIndicator', () => {
    it('should be rendered correctly', () => {
        const { asFragment } = render(<StatusIndicator caption="Indicator" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const { asFragment } = render(
            <StatusIndicator caption="Indicator" color="info" fill="outline" size="12" />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
