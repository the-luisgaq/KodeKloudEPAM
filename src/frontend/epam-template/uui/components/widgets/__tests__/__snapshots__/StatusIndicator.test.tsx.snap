// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIndicator should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="root uui-size-24 uui-status_indicator uui-color-neutral uui-fill-solid"
  >
    <div
      class="uui-status_indicator-dot"
    />
    <p
      class="uui-status_indicator-caption"
    >
      Indicator
    </p>
  </div>
</DocumentFragment>
`;

exports[`StatusIndicator should be rendered correctly with props 1`] = `
<DocumentFragment>
  <div
    class="root uui-size-12 uui-status_indicator uui-color-info uui-fill-outline"
  >
    <div
      class="uui-status_indicator-dot"
    />
    <p
      class="uui-status_indicator-caption"
    >
      Indicator
    </p>
  </div>
</DocumentFragment>
`;
